pkg/bytecode/compiler.go

```diff
@@ -286,6 +293,50 @@ func (c *Compiler) compileConditionalBlock(block *parser.ConditionalBlock) (int,
 	return jumpPos, nil
 }
 
+func (c *Compiler) compileWhileStatement(stmt *parser.WhileStmt) error {
+	startPos := len(c.instructions)
+	if err := c.Compile(stmt.Condition); err != nil {
+		return err
+	}
+	// Prepare end position of while block, jump to end if
+	jumpOnFalsePos, err := c.emitPos(OpJumpOnFalse, JumpPlaceholder)
+	if err != nil {
+		return err
+	}
+	// take a snapshot of the break list before compiling the body of the
+	// loop
+	outOfScopeBreaks := c.flushBreaks()
```

Comment:
Since flushBreaks does not need to copy (other comment), I think  you can get rid of it and make this:
```
outOfScopeBreaks := c.breaks
c.breaks = []int{}
```